### DISCUSSION TITLE
ERROR: missing FROM-clause entry for table "raw_user_meta_data" on trigger function

### DISCUSSION THREAD

**Discussion by elte0** [supabase/supabase#3787] [Questions] [answered]
I'm trying to create a trigger function that creates a profile on user sign up according to [this](https://supabase.io/docs/guides/auth/managing-user-data#advanced-techniques) guide.

My trigger function looks like this:
```sql
create function public.handle_new_user() 
returns trigger 
language plpgsql 
security definer set search_path = public
as $$
begin
  insert into public.profiles (id, first_name)
  values (new.id, new.raw_user_meta_data.first_name);
  return new;
end;
$$;

-- trigger the function every time a user is created
create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();

```

The insertion in client fails with error:
```js
Object {
  "message": "Database error saving new user",
  "status": 500,
}
```

And in the Postgres log it shows:
```
ERROR: missing FROM-clause entry for table "raw_user_meta_data"
```

I've spent some time on this now and don't understand the error?

Can someone explain?

**Comment by elte0** [comment, marked as answer]
**Fixed:**
I had to replace `new.raw_user_meta_data.first_name` with `new.raw_user_meta_data->>'first_name'`

**Comment by EskelCz** [comment]
In my case it was missing a join that seemed unnecessary, as it worked in the sql editor without it. Turns out in pgplsql it was needed. Weird
